tests/cases/compiler/useUnknownInCatchVariables02.ts(2,10): error TS2339: Property 'name' does not exist on type 'unknown'.
tests/cases/compiler/useUnknownInCatchVariables02.ts(5,10): error TS2339: Property 'name' does not exist on type 'unknown'.
tests/cases/compiler/useUnknownInCatchVariables02.ts(8,10): error TS2339: Property 'a' does not exist on type 'unknown'.
tests/cases/compiler/useUnknownInCatchVariables02.ts(11,10): error TS2339: Property 'a' does not exist on type 'unknown'.
tests/cases/compiler/useUnknownInCatchVariables02.ts(14,8): error TS2461: Type 'unknown' is not an array type.
tests/cases/compiler/useUnknownInCatchVariables02.ts(17,8): error TS2461: Type 'unknown' is not an array type.


==== tests/cases/compiler/useUnknownInCatchVariables02.ts (6 errors) ====
    try {}
    catch ({ name }) { name; }
             ~~~~
!!! error TS2339: Property 'name' does not exist on type 'unknown'.
    
    try {}
    catch ({ name }) {}
             ~~~~
!!! error TS2339: Property 'name' does not exist on type 'unknown'.
    
    try {}
    catch ({ a: { b } }) { b; }
             ~
!!! error TS2339: Property 'a' does not exist on type 'unknown'.
    
    try {}
    catch ({ a: { b } }) {}
             ~
!!! error TS2339: Property 'a' does not exist on type 'unknown'.
    
    try {}
    catch ([a, { b }]) { a; }
           ~~~~~~~~~~
!!! error TS2461: Type 'unknown' is not an array type.
    
    try {}
    catch ([a, { b }]) {}
           ~~~~~~~~~~
!!! error TS2461: Type 'unknown' is not an array type.
    